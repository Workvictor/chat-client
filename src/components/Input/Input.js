import React from 'react';
import styled from 'styled-components';


const Wrapper = styled.input`
  background: #fff;
  color: #333;
  border: 1px solid #999;
`

export class Input extends React.Component {
  state = {
    value: ``,
  }

  onUserInput = ({ value }) => {
    const { onChange } = this.props;
    this.onValueReceive({ value });
    onChange && onChange({ value });
  }

  onValueReceive = ({ value }) => {
    value !== this.state.value && this.setState({ value });
  }

  componentDidMount(){
    const { value } = this.props;
    value && this.onValueReceive({ value });
  }

  componentWillReceiveProps(nextProps) {
    const { value } = nextProps;
    value !== this.props.value && this.onValueReceive({ value });
  }

  render() {
    const { value } = this.state;
    return (
      <Wrapper {...this.state} onChange={this.onUserInput} />
    )
  }
}