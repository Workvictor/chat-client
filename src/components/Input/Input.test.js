import React from 'react';
import { shallow, mount } from 'enzyme';
import { Input } from './Input';


describe(`Компонент Input`, () => {
  test(`должен соответствовать снимку`, () => {
    const snapshot = mount(<Input value={`test`} />);
    expect(snapshot).toMatchSnapshot();
  })

  it(`должен рендерится без ошибок`, () => {
    shallow(<Input />);
  })

  it(`должен изменять значение при изменении состояния`, () => {
    const input = mount(<Input />);
    input.setState({ value: `test` });
    expect(input.state(`value`)).toBe(`test`);
  })

  it(`должен изменять состояние при монтировании, если задано начальное знчение`, () => {
    const newValue = `newValue`;
    const input = shallow(<Input value={newValue} />);
    expect(input.state(`value`)).toBe(newValue);
  })

  it(`должен изменять значение при получении нового значения, если оно отличается от текущего`, () => {
    const newValue = `newValue`;
    const input = shallow(<Input />);
    input.setProps({ value: newValue });
    expect(input.state(`value`)).toBe(newValue);
  })

  it(`должен изменять значение состояние при пользовательском вводе`, ()=>{
    const userInput = `newValue`;
    const input = mount(<Input />);
    input.simulate(`change`, {value: userInput});
    expect(input.state(`value`)).toBe(userInput);
  })

})
