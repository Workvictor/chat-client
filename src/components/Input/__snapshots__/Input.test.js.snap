// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Input должен соответствовать снимку 1`] = `
<Input
  value="test"
>
  <styled.input
    onChange={[Function]}
    value="test"
  >
    <input
      className="sc-bdVaJa bxeEtt"
      onChange={[Function]}
      value="test"
    />
  </styled.input>
</Input>
`;
